{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C:\\Users\\johnp\\Downloads\\poly-a_loop\\reverse_strand\n"
     ]
    }
   ],
   "source": [
    "from pathlib import Path\n",
    "import glob\n",
    "import os\n",
    "dirt  =  %pwd\n",
    "print(dirt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unique reads mapped to Reverse strand \n",
      "single_hits_minus_strand_e1-05_soc1_blastn_bd_only_1348_50-1_epat   685\n",
      "single_hits_minus_strand_e1-05_soc1_blastn_bd_only_1348_50-2_epat   803\n",
      "single_hits_minus_strand_e1-05_soc1_blastn_bd_only_1348_67-1_epat   883\n",
      "single_hits_minus_strand_e1-05_soc1_blastn_bd_only_1348_67-2_epat   1217\n",
      "single_hits_minus_strand_e1-05_soc1_blastn_bd_only_1348_69-1_epat   595\n",
      "single_hits_minus_strand_e1-05_soc1_blastn_bd_only_1348_69-2_epat   562\n",
      "single_hits_minus_strand_e1-05_soc1_blastn_bd_w_deadenylase_1349_46-1_epat   613\n",
      "single_hits_minus_strand_e1-05_soc1_blastn_bd_w_deadenylase_1349_46-2_epat   584\n",
      "single_hits_minus_strand_e1-05_soc1_blastn_bd_w_deadenylase_1349_58-1_epat   612\n",
      "single_hits_minus_strand_e1-05_soc1_blastn_bd_w_deadenylase_1349_58-2_epat   788\n",
      "single_hits_minus_strand_e1-05_soc1_blastn_bd_w_deadenylase_1349_68-1_epat   589\n",
      "single_hits_minus_strand_e1-05_soc1_blastn_bd_w_deadenylase_1349_68-2_epat   950\n",
      "single_hits_minus_strand_e1-05_soc1_blastn_tvn_1348-1_epat   677\n",
      "single_hits_minus_strand_e1-05_soc1_blastn_tvn_1348-2_epat   497\n",
      "single_hits_minus_strand_e1-05_soc1_blastn_wt_37-1_epat   1694\n",
      "single_hits_minus_strand_e1-05_soc1_blastn_wt_37-2_epat   645\n",
      "single_hits_minus_strand_e1-05_soc1_blastn_wt_50-1_epat   1608\n",
      "single_hits_minus_strand_e1-05_soc1_blastn_wt_50-2_epat   2214\n",
      "single_hits_minus_strand_e1-05_soc1_blastn_wt_54-1_epat   1149\n",
      "single_hits_minus_strand_e1-05_soc1_blastn_wt_54-2_epat   962\n"
     ]
    }
   ],
   "source": [
    "count_minus = 0\n",
    "lst_plus = []\n",
    "lst_minus = []\n",
    "print(\"Unique reads mapped to Reverse strand \")\n",
    "for filename in glob.glob('Genotypes/*.txt'):\n",
    "    x = os.path.splitext(os.path.basename(filename))[0]\n",
    "    directory = x\n",
    "    file = open(filename,\"r\")\n",
    "    for line in file:\n",
    "        line = line.split()\n",
    "        x = int(line[8])\n",
    "        y = int(line[9])\n",
    "        if (x>y):\n",
    "            count_minus = count_minus + 1\n",
    "        else:\n",
    "            pass\n",
    "    print(directory, \" \", count_minus)\n",
    "    lst_plus.append(count_minus)\n",
    "    count_minus = 0\n",
    "\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "None\n",
      "One 50\n",
      "Two 18\n",
      "Three 55\n",
      "Four 20\n",
      "Five 195\n",
      "Six 207\n",
      "Seven 37\n",
      "Eight 10\n",
      "*********End of Genotype******\n",
      "None\n",
      "One 95\n",
      "Two 44\n",
      "Three 135\n",
      "Four 40\n",
      "Five 439\n",
      "Six 398\n",
      "Seven 66\n",
      "Eight 22\n",
      "*********End of Genotype******\n",
      "None\n",
      "One 170\n",
      "Two 80\n",
      "Three 233\n",
      "Four 69\n",
      "Five 650\n",
      "Six 678\n",
      "Seven 107\n",
      "Eight 35\n",
      "*********End of Genotype******\n",
      "None\n",
      "One 272\n",
      "Two 116\n",
      "Three 349\n",
      "Four 103\n",
      "Five 967\n",
      "Six 1010\n",
      "Seven 173\n",
      "Eight 54\n",
      "*********End of Genotype******\n",
      "None\n",
      "One 321\n",
      "Two 134\n",
      "Three 411\n",
      "Four 122\n",
      "Five 1108\n",
      "Six 1196\n",
      "Seven 206\n",
      "Eight 63\n",
      "*********End of Genotype******\n",
      "None\n",
      "One 355\n",
      "Two 150\n",
      "Three 473\n",
      "Four 139\n",
      "Five 1253\n",
      "Six 1357\n",
      "Seven 231\n",
      "Eight 73\n",
      "*********End of Genotype******\n",
      "None\n",
      "One 396\n",
      "Two 159\n",
      "Three 510\n",
      "Four 157\n",
      "Five 1402\n",
      "Six 1545\n",
      "Seven 256\n",
      "Eight 84\n",
      "*********End of Genotype******\n",
      "None\n",
      "One 426\n",
      "Two 167\n",
      "Three 551\n",
      "Four 169\n",
      "Five 1539\n",
      "Six 1691\n",
      "Seven 279\n",
      "Eight 90\n",
      "*********End of Genotype******\n",
      "None\n",
      "One 464\n",
      "Two 178\n",
      "Three 610\n",
      "Four 184\n",
      "Five 1705\n",
      "Six 1860\n",
      "Seven 306\n",
      "Eight 99\n",
      "*********End of Genotype******\n",
      "None\n",
      "One 514\n",
      "Two 202\n",
      "Three 663\n",
      "Four 208\n",
      "Five 1941\n",
      "Six 2033\n",
      "Seven 336\n",
      "Eight 114\n",
      "*********End of Genotype******\n",
      "None\n",
      "One 558\n",
      "Two 219\n",
      "Three 708\n",
      "Four 224\n",
      "Five 2075\n",
      "Six 2183\n",
      "Seven 353\n",
      "Eight 125\n",
      "*********End of Genotype******\n",
      "None\n",
      "One 603\n",
      "Two 243\n",
      "Three 772\n",
      "Four 242\n",
      "Five 2291\n",
      "Six 2367\n",
      "Seven 384\n",
      "Eight 135\n",
      "*********End of Genotype******\n",
      "None\n",
      "One 661\n",
      "Two 295\n",
      "Three 817\n",
      "Four 261\n",
      "Five 2439\n",
      "Six 2612\n",
      "Seven 410\n",
      "Eight 156\n",
      "*********End of Genotype******\n",
      "None\n",
      "One 698\n",
      "Two 321\n",
      "Three 860\n",
      "Four 277\n",
      "Five 2578\n",
      "Six 2763\n",
      "Seven 424\n",
      "Eight 170\n",
      "*********End of Genotype******\n",
      "None\n",
      "One 846\n",
      "Two 377\n",
      "Three 1049\n",
      "Four 342\n",
      "Five 2983\n",
      "Six 3304\n",
      "Seven 531\n",
      "Eight 202\n",
      "*********End of Genotype******\n",
      "None\n",
      "One 910\n",
      "Two 393\n",
      "Three 1114\n",
      "Four 361\n",
      "Five 3129\n",
      "Six 3516\n",
      "Seven 566\n",
      "Eight 210\n",
      "*********End of Genotype******\n",
      "None\n",
      "One 1073\n",
      "Two 440\n",
      "Three 1303\n",
      "Four 426\n",
      "Five 3477\n",
      "Six 4083\n",
      "Seven 666\n",
      "Eight 246\n",
      "*********End of Genotype******\n",
      "None\n",
      "One 1288\n",
      "Two 495\n",
      "Three 1539\n",
      "Four 497\n",
      "Five 4075\n",
      "Six 4762\n",
      "Seven 798\n",
      "Eight 284\n",
      "*********End of Genotype******\n",
      "None\n",
      "One 1381\n",
      "Two 543\n",
      "Three 1660\n",
      "Four 541\n",
      "Five 4366\n",
      "Six 5108\n",
      "Seven 865\n",
      "Eight 301\n",
      "*********End of Genotype******\n",
      "None\n",
      "One 1447\n",
      "Two 568\n",
      "Three 1783\n",
      "Four 573\n",
      "Five 4595\n",
      "Six 5382\n",
      "Seven 916\n",
      "Eight 319\n",
      "*********End of Genotype******\n"
     ]
    }
   ],
   "source": [
    "#### Getting ids for all ten categories\n",
    "count = []\n",
    "index_1,index_2,index_3,index_4,index_5,index_6,index_7,index_8,index_9,index_10 = 0,0,0,0,0,0,0,0,0,0\n",
    "for filename in glob.glob('Genotypes/*.txt'):\n",
    "    file_open = open(filename,\"r\")\n",
    "    x = os.path.splitext(os.path.basename(filename))[0]\n",
    "    directory = x\n",
    "    parent_dir = dirt + \"/Genotypes\"\n",
    "    path = os.path.join(parent_dir, directory) \n",
    "    new_path = os.mkdir(path) \n",
    "    sub_ids = \"ids\"\n",
    "    path_ids = os.path.join(path, sub_ids)\n",
    "    ids_path = os.mkdir(path_ids)\n",
    "    print(ids_path)\n",
    "    sub_fasta = \"fasta\"\n",
    "    path_fasta = os.path.join(path, sub_fasta)\n",
    "    os.mkdir(path_fasta)\n",
    "    for i in range(1,9):\n",
    "#         if i == 1:\n",
    "#             ids_path =  path_ids + \"\\Category_1.txt\"\n",
    "#             re1 = open(ids_path,\"a\") \n",
    "#             for line in file_open:\n",
    "#                 line = line.split()\n",
    "#                 x = int(line[8])\n",
    "#                 y = int(line[9])\n",
    "#                 if (x>=1) and (y<=24):\n",
    "#                     re1.write((\">\"+line[0]))\n",
    "#                     re1.write(\"\\n\")\n",
    "#                     index_1 = index_1 + 1\n",
    "#             print(\"One\",index_1)\n",
    "#             re1.close()\n",
    "\n",
    "\n",
    "            \n",
    "        if i == 1:\n",
    "            ids_path =  path_ids + \"\\Category_1.txt\"\n",
    "            re2 = open(ids_path,\"a\")\n",
    "            file_open = open(filename,\"r\")\n",
    "            for line in file_open:\n",
    "                line = line.split()\n",
    "                x = int(line[8])\n",
    "                y = int(line[9])\n",
    "                if (y>=25) and (x<=100):\n",
    "                    re2.write((\">\"+line[0]))\n",
    "                    re2.write(\"\\n\")\n",
    "                    index_2 = index_2 + 1\n",
    "            print(\"One\",index_2)\n",
    "\n",
    "            re2.close()\n",
    "\n",
    "\n",
    "#         elif i == 3:\n",
    "#             ids_path =  path_ids + \"\\Category_3.txt\"\n",
    "#             re3 = open(ids_path,\"a\") \n",
    "#             file_open = open(filename,\"r\")\n",
    "#             for line in file_open:\n",
    "#                 line = line.split()\n",
    "#                 x = int(line[8])\n",
    "#                 y = int(line[9])\n",
    "#                 if (x>=101) and (y<=107):\n",
    "#                     re3.write((\">\"+line[0]))\n",
    "#                     re3.write(\"\\n\")\n",
    "#                     index_3 = index_3 + 1\n",
    "#             print(\"Three\",index_3)\n",
    "#             re3.close()\n",
    "\n",
    "        elif i == 2:\n",
    "            ids_path =  path_ids + \"\\Category_2.txt\"\n",
    "            re4 = open(ids_path,\"a\") \n",
    "            file_open = open(filename,\"r\")\n",
    "            for line in file_open:\n",
    "                line = line.split()\n",
    "                x = int(line[8])\n",
    "                y = int(line[9])\n",
    "                if (y>=108) and (x<=329):\n",
    "                    re4.write((\">\"+line[0]))\n",
    "                    re4.write(\"\\n\")\n",
    "                    index_4 = index_4 + 1\n",
    "            print(\"Two\",index_4)\n",
    "            re4.close()\n",
    "\n",
    "\n",
    "        elif i == 3:\n",
    "            ids_path =  path_ids + \"\\Category_3.txt\"\n",
    "            re5 = open(ids_path,\"a\") \n",
    "            file_open = open(filename,\"r\")\n",
    "            for line in file_open:\n",
    "                line = line.split()\n",
    "                x = int(line[8])\n",
    "                y = int(line[9])\n",
    "                if (y<=24) and (108<=x<=329):\n",
    "                    re5.write((\">\"+line[0]))\n",
    "                    re5.write(\"\\n\")\n",
    "                    index_5 = index_5 + 1\n",
    "            print(\"Three\",index_5)\n",
    "            re5.close()\n",
    "\n",
    "\n",
    "        elif i == 4:\n",
    "            ids_path =  path_ids + \"\\Category_4.txt\"\n",
    "            re6 = open(ids_path,\"a\") \n",
    "            file_open = open(filename,\"r\")\n",
    "            for line in file_open:\n",
    "                line = line.split()\n",
    "                x = int(line[8])\n",
    "                y = int(line[9])\n",
    "                if (y<=24) and (101<=x<=107):\n",
    "                    re6.write((\">\"+line[0]))\n",
    "                    re6.write(\"\\n\")\n",
    "                    index_6 = index_6 + 1\n",
    "            print(\"Four\",index_6)\n",
    "            re4.close()\n",
    "\n",
    "\n",
    "        elif i == 5:\n",
    "            ids_path =  path_ids + \"\\Category_5.txt\"\n",
    "            re7 = open(ids_path,\"a\")\n",
    "            file_open = open(filename,\"r\")\n",
    "            for line in file_open:\n",
    "                line = line.split()\n",
    "                x = int(line[8])\n",
    "                y = int(line[9])\n",
    "                if (y<=24) and (25<=x<=100):\n",
    "                    re7.write((\">\"+line[0]))\n",
    "                    re7.write(\"\\n\")\n",
    "                    index_7 = index_7 + 1\n",
    "            print(\"Five\",index_7)\n",
    "            re7.close()\n",
    "\n",
    "\n",
    "        elif i == 6:\n",
    "            ids_path =  path_ids + \"\\Category_6.txt\"\n",
    "            re8 = open(ids_path,\"a\")\n",
    "            file_open = open(filename,\"r\")\n",
    "            for line in file_open:\n",
    "                line = line.split()\n",
    "                x = int(line[8])\n",
    "                y = int(line[9])\n",
    "                if (25<=y<=100) and (108<=x<=329):\n",
    "                    re8.write((\">\"+line[0]))\n",
    "                    re8.write(\"\\n\")\n",
    "                    index_8 = index_8 + 1\n",
    "            print(\"Six\",index_8)\n",
    "            re8.close()\n",
    "\n",
    "\n",
    "        elif i == 7:\n",
    "            ids_path =  path_ids + \"\\Category_7.txt\"\n",
    "            re9 = open(ids_path,\"a\")\n",
    "            file_open = open(filename,\"r\")\n",
    "            for line in file_open:\n",
    "                line = line.split()\n",
    "                x = int(line[8])\n",
    "                y = int(line[9])\n",
    "                if (25<=y<=100) and (101<=x<=107):\n",
    "                    re9.write((\">\"+line[0]))\n",
    "                    re9.write(\"\\n\")\n",
    "                    index_9 = index_9 + 1\n",
    "            print(\"Seven\",index_9)\n",
    "            re9.close()\n",
    "\n",
    "\n",
    "        elif i == 8:\n",
    "            ids_path =  path_ids + \"\\Category_8.txt\"\n",
    "            re10 = open(ids_path,\"a\")\n",
    "            file_open = open(filename,\"r\")\n",
    "            for line in file_open:\n",
    "                line = line.split()\n",
    "                x = int(line[8])\n",
    "                y = int(line[9])\n",
    "                if (101<=y<=107) and (108<=x<=329):\n",
    "                    re10.write((\">\"+line[0]))\n",
    "                    re10.write(\"\\n\")\n",
    "                    index_10 = index_10 + 1\n",
    "            print(\"Eight\",index_10)\n",
    "            re10.close()\n",
    "            print(\"*********End of Genotype******\")\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Getting fasta sequences in a loop\n",
    "def getting_fasta(genotype,sequence_file):\n",
    "    main_path = 'C:/Users/johnp/Downloads/poly-a_loop/reverse_strand/Genotypes/'\n",
    "    for i in range(1,9):\n",
    "\n",
    "        filename  = main_path + genotype +\"/ids/Category_%i.txt\"%i\n",
    "        file_ids = open(filename,\"r\")\n",
    "\n",
    "        filename_fasta = main_path + genotype + \"/fasta/Category_%i.txt\"%i\n",
    "        re = open(filename_fasta,\"a\")           \n",
    "        ids = []\n",
    "        fasta = []\n",
    "        for line in file_ids: ### Working with sequence header only\n",
    "            line = line.strip(\"\\n\")\n",
    "            ids.append(line)\n",
    "        # print(ids)\n",
    "        main_path_to_sequence_file = \"C:/Users/johnp/Downloads/poly-a_loop/reverse_strand/intial_fasta_files/\"\n",
    "        sequence_path =  main_path_to_sequence_file + sequence_file\n",
    "        file_fasta = open(sequence_path,\"r\") ### Tuple with id and seq\n",
    "        ids_seq_list = []\n",
    "        for line2 in file_fasta:\n",
    "            words = line2.split()\n",
    "            if line2[0] == \">\":\n",
    "                    y = line2\n",
    "            else:\n",
    "                    line2 = line2.strip(\"\\n\") \n",
    "                    ids_seq_list.append((y.strip(\"\\n\"),line2))\n",
    "        # print(ids_seq_list)\n",
    "\n",
    "        ## Getting fasta sequences after comparing ids\n",
    "        for x in ids:\n",
    "        #     print(\"Working\")\n",
    "            for y in ids_seq_list:\n",
    "        #         print(\"Working\")\n",
    "                y1 = y[0]\n",
    "                if (x == y1):\n",
    "    #                 print(x + \"\\n\" + y[1])\n",
    "                    re.write(x)\n",
    "                    re.write(\"\\n\")\n",
    "                    re.write(y[1])\n",
    "                    re.write(\"\\n\")\n",
    "\n",
    "    re.close()\n",
    "    \n",
    "    return 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "getting_fasta(\"single_hits_minus_strand_e1-05_soc1_blastn_bd_only_1348_50-1_epat\",\"refine_final_1.fa\")\n",
    "getting_fasta(\"single_hits_minus_strand_e1-05_soc1_blastn_bd_only_1348_50-2_epat\",\"refine_final_2.fa\")\n",
    "getting_fasta(\"single_hits_minus_strand_e1-05_soc1_blastn_bd_only_1348_67-1_epat\",\"refine_final_3.fa\")\n",
    "getting_fasta(\"single_hits_minus_strand_e1-05_soc1_blastn_bd_only_1348_67-2_epat\",\"refine_final_4.fa\")\n",
    "getting_fasta(\"single_hits_minus_strand_e1-05_soc1_blastn_bd_only_1348_69-1_epat\",\"refine_final_5.fa\")\n",
    "getting_fasta(\"single_hits_minus_strand_e1-05_soc1_blastn_bd_only_1348_69-2_epat\",\"refine_final_6.fa\")\n",
    "getting_fasta(\"single_hits_minus_strand_e1-05_soc1_blastn_bd_w_deadenylase_1349_46-1_epat\",\"refine_final_7.fa\")\n",
    "getting_fasta(\"single_hits_minus_strand_e1-05_soc1_blastn_bd_w_deadenylase_1349_46-2_epat\",\"refine_final_8.fa\")\n",
    "getting_fasta(\"single_hits_minus_strand_e1-05_soc1_blastn_bd_w_deadenylase_1349_58-1_epat\",\"refine_final_9.fa\")\n",
    "getting_fasta(\"single_hits_minus_strand_e1-05_soc1_blastn_bd_w_deadenylase_1349_58-2_epat\",\"refine_final_10.fa\")\n",
    "getting_fasta(\"single_hits_minus_strand_e1-05_soc1_blastn_bd_w_deadenylase_1349_68-1_epat\",\"refine_final_11.fa\")\n",
    "getting_fasta(\"single_hits_minus_strand_e1-05_soc1_blastn_bd_w_deadenylase_1349_68-2_epat\",\"refine_final_12.fa\")\n",
    "getting_fasta(\"single_hits_minus_strand_e1-05_soc1_blastn_tvn_1348-1_epat\",\"refine_final_13.fa\")\n",
    "getting_fasta(\"single_hits_minus_strand_e1-05_soc1_blastn_tvn_1348-2_epat\",\"refine_final_14.fa\")\n",
    "getting_fasta(\"single_hits_minus_strand_e1-05_soc1_blastn_wt_37-1_epat\",\"refine_final_15.fa\")\n",
    "getting_fasta(\"single_hits_minus_strand_e1-05_soc1_blastn_wt_37-2_epat\",\"refine_final_16.fa\")\n",
    "getting_fasta(\"single_hits_minus_strand_e1-05_soc1_blastn_wt_50-1_epat\",\"refine_final_17.fa\")\n",
    "getting_fasta(\"single_hits_minus_strand_e1-05_soc1_blastn_wt_50-2_epat\",\"refine_final_18.fa\")\n",
    "getting_fasta(\"single_hits_minus_strand_e1-05_soc1_blastn_wt_54-1_epat\",\"refine_final_19.fa\")\n",
    "getting_fasta(\"single_hits_minus_strand_e1-05_soc1_blastn_wt_54-2_epat\",\"refine_final_20.fa\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
