{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C:\\Users\\johnp\\Downloads\\poly-a_loop\\forward_strand\n"
     ]
    }
   ],
   "source": [
    "## Importing required libraries\n",
    "from pathlib import Path\n",
    "import glob\n",
    "import os\n",
    "dirt  =  %pwd ## Getting current working directory\n",
    "print(dirt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unique reads mapped to positive strand\n",
      "singel_hits_plus_strand_e1-05_soc1_blastn_bd_only_1348_50-1_epat   581\n",
      "singel_hits_plus_strand_e1-05_soc1_blastn_bd_only_1348_50-2_epat   756\n",
      "singel_hits_plus_strand_e1-05_soc1_blastn_bd_only_1348_67-1_epat   836\n",
      "singel_hits_plus_strand_e1-05_soc1_blastn_bd_only_1348_67-2_epat   1144\n",
      "singel_hits_plus_strand_e1-05_soc1_blastn_bd_only_1348_69-1_epat   522\n",
      "singel_hits_plus_strand_e1-05_soc1_blastn_bd_only_1348_69-2_epat   598\n",
      "singel_hits_plus_strand_e1-05_soc1_blastn_bd_w_deadenylase_1349_46-1_epat   569\n",
      "singel_hits_plus_strand_e1-05_soc1_blastn_bd_w_deadenylase_1349_46-2_epat   615\n",
      "singel_hits_plus_strand_e1-05_soc1_blastn_bd_w_deadenylase_1349_58-1_epat   574\n",
      "singel_hits_plus_strand_e1-05_soc1_blastn_bd_w_deadenylase_1349_58-2_epat   783\n",
      "singel_hits_plus_strand_e1-05_soc1_blastn_bd_w_deadenylase_1349_68-1_epat   588\n",
      "singel_hits_plus_strand_e1-05_soc1_blastn_bd_w_deadenylase_1349_68-2_epat   875\n",
      "singel_hits_plus_strand_e1-05_soc1_blastn_tvn_1348-1_epat   599\n",
      "singel_hits_plus_strand_e1-05_soc1_blastn_tvn_1348-2_epat   436\n",
      "singel_hits_plus_strand_e1-05_soc1_blastn_wt_37-1_epat   1439\n",
      "singel_hits_plus_strand_e1-05_soc1_blastn_wt_37-2_epat   570\n",
      "singel_hits_plus_strand_e1-05_soc1_blastn_wt_50-1_epat   1457\n",
      "singel_hits_plus_strand_e1-05_soc1_blastn_wt_50-2_epat   1875\n",
      "singel_hits_plus_strand_e1-05_soc1_blastn_wt_54-1_epat   1104\n",
      "singel_hits_plus_strand_e1-05_soc1_blastn_wt_54-2_epat   906\n"
     ]
    }
   ],
   "source": [
    "count_plus = 0\n",
    "count_minus = 0\n",
    "lst_plus = []\n",
    "lst_minus = []\n",
    "print(\"Unique reads mapped to positive strand\")\n",
    "for filename in glob.glob('Genotypes/*.txt'): ## Getting each file (.txt) from the Gebotypes folder\n",
    "    x = os.path.splitext(os.path.basename(filename))[0] ## Getting only the genotype name withput .txt extension\n",
    "    name = x\n",
    "    file = open(filename,\"r\") ## Opening the file\n",
    "    for line in file:\n",
    "        line = line.split()\n",
    "        x = int(line[8])\n",
    "        y = int(line[9])\n",
    "        if (x<y):\n",
    "            count_plus = count_plus + 1 ## Knowing the total hits in each genotype file\n",
    "        else:\n",
    "            pass\n",
    "    print(name,' ',count_plus)         ## Printing total hits in each file\n",
    "    lst_plus.append(count_plus)\n",
    "    count_plus = 0\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Two 38\n",
      "Four 12\n",
      "Five 42\n",
      "Six 17\n",
      "Seven 169\n",
      "Eight 147\n",
      "Nine 24\n",
      "Ten 8\n",
      "*********End of Genotype******\n",
      "Two 80\n",
      "Four 31\n",
      "Five 98\n",
      "Six 40\n",
      "Seven 422\n",
      "Eight 326\n",
      "Nine 43\n",
      "Ten 11\n",
      "*********End of Genotype******\n",
      "Two 127\n",
      "Four 63\n",
      "Five 191\n",
      "Six 58\n",
      "Seven 648\n",
      "Eight 570\n",
      "Nine 83\n",
      "Ten 21\n",
      "*********End of Genotype******\n",
      "Two 200\n",
      "Four 98\n",
      "Five 297\n",
      "Six 86\n",
      "Seven 958\n",
      "Eight 852\n",
      "Nine 120\n",
      "Ten 34\n",
      "*********End of Genotype******\n",
      "Two 230\n",
      "Four 108\n",
      "Five 362\n",
      "Six 95\n",
      "Seven 1108\n",
      "Eight 992\n",
      "Nine 141\n",
      "Ten 42\n",
      "*********End of Genotype******\n",
      "Two 263\n",
      "Four 128\n",
      "Five 423\n",
      "Six 112\n",
      "Seven 1259\n",
      "Eight 1135\n",
      "Nine 160\n",
      "Ten 54\n",
      "*********End of Genotype******\n",
      "Two 296\n",
      "Four 136\n",
      "Five 478\n",
      "Six 129\n",
      "Seven 1409\n",
      "Eight 1258\n",
      "Nine 174\n",
      "Ten 63\n",
      "*********End of Genotype******\n",
      "Two 317\n",
      "Four 147\n",
      "Five 516\n",
      "Six 142\n",
      "Seven 1602\n",
      "Eight 1364\n",
      "Nine 184\n",
      "Ten 70\n",
      "*********End of Genotype******\n",
      "Two 351\n",
      "Four 158\n",
      "Five 567\n",
      "Six 158\n",
      "Seven 1753\n",
      "Eight 1515\n",
      "Nine 210\n",
      "Ten 79\n",
      "*********End of Genotype******\n",
      "Two 386\n",
      "Four 178\n",
      "Five 628\n",
      "Six 179\n",
      "Seven 2001\n",
      "Eight 1676\n",
      "Nine 250\n",
      "Ten 93\n",
      "*********End of Genotype******\n",
      "Two 401\n",
      "Four 194\n",
      "Five 672\n",
      "Six 189\n",
      "Seven 2163\n",
      "Eight 1793\n",
      "Nine 266\n",
      "Ten 100\n",
      "*********End of Genotype******\n",
      "Two 421\n",
      "Four 208\n",
      "Five 730\n",
      "Six 203\n",
      "Seven 2417\n",
      "Eight 1931\n",
      "Nine 281\n",
      "Ten 106\n",
      "*********End of Genotype******\n",
      "Two 453\n",
      "Four 260\n",
      "Five 778\n",
      "Six 216\n",
      "Seven 2566\n",
      "Eight 2119\n",
      "Nine 302\n",
      "Ten 125\n",
      "*********End of Genotype******\n",
      "Two 474\n",
      "Four 286\n",
      "Five 809\n",
      "Six 226\n",
      "Seven 2684\n",
      "Eight 2245\n",
      "Nine 319\n",
      "Ten 139\n",
      "*********End of Genotype******\n",
      "Two 588\n",
      "Four 330\n",
      "Five 960\n",
      "Six 266\n",
      "Seven 3091\n",
      "Eight 2662\n",
      "Nine 390\n",
      "Ten 165\n",
      "*********End of Genotype******\n",
      "Two 621\n",
      "Four 350\n",
      "Five 1027\n",
      "Six 283\n",
      "Seven 3244\n",
      "Eight 2817\n",
      "Nine 419\n",
      "Ten 178\n",
      "*********End of Genotype******\n",
      "Two 727\n",
      "Four 389\n",
      "Five 1197\n",
      "Six 338\n",
      "Seven 3669\n",
      "Eight 3245\n",
      "Nine 505\n",
      "Ten 203\n",
      "*********End of Genotype******\n",
      "Two 869\n",
      "Four 435\n",
      "Five 1381\n",
      "Six 405\n",
      "Seven 4222\n",
      "Eight 3797\n",
      "Nine 592\n",
      "Ten 235\n",
      "*********End of Genotype******\n",
      "Two 951\n",
      "Four 465\n",
      "Five 1478\n",
      "Six 447\n",
      "Seven 4544\n",
      "Eight 4118\n",
      "Nine 631\n",
      "Ten 246\n",
      "*********End of Genotype******\n",
      "Two 1011\n",
      "Four 489\n",
      "Five 1573\n",
      "Six 471\n",
      "Seven 4821\n",
      "Eight 4366\n",
      "Nine 666\n",
      "Ten 265\n",
      "*********End of Genotype******\n"
     ]
    }
   ],
   "source": [
    "#### Getting ids for all ten categories\n",
    "count = []\n",
    "index_1,index_2,index_3,index_4,index_5,index_6,index_7,index_8 = 0,0,0,0,0,0,0,0\n",
    "for filename in glob.glob('Genotypes/*.txt'): ## Getting each file from the genotype folder\n",
    "    file_open = open(filename,\"r\") ### OPening the folder\n",
    "    x = os.path.splitext(os.path.basename(filename))[0] ## ## Getting only the genotype name withput .txt extension\n",
    "    directory = x\n",
    "    parent_dir = dirt + \"/Genotypes\"\n",
    "    path = os.path.join(parent_dir, directory) \n",
    "    new_path = os.mkdir(path)  ## Creating a folder with the genotype name\n",
    "    sub_ids = \"ids\" \n",
    "    path_ids = os.path.join(path, sub_ids)\n",
    "    ids_path = os.mkdir(path_ids) # Creating the folder \"Ids\" to store ids for each category (8 categories in total)\n",
    "    sub_fasta = \"fasta\"\n",
    "    path_fasta = os.path.join(path, sub_fasta)\n",
    "    os.mkdir(path_fasta) # Creating the folder \"fasta\" to store fasta sequences of the specific ids for specific gebotype\n",
    "    for i in range(1,9): ## As total categories are 8, we are looping 8 times\n",
    "        \n",
    "#         if i == 1:\n",
    "#             ids_path =  path_ids + \"\\Category_1.txt\"\n",
    "#             re1 = open(ids_path,\"a\") \n",
    "#             for line in file_open:\n",
    "#                 line = line.split()\n",
    "#                 x = int(line[8])\n",
    "#                 y = int(line[9])\n",
    "#                 if (x>=1) and (y<=24):\n",
    "#                     re1.write((\">\"+line[0]))\n",
    "#                     re1.write(\"\\n\")\n",
    "#                     index_1 = index_1 + 1\n",
    "#             print(\"One\",index_1)\n",
    "#             re1.close()\n",
    "\n",
    "\n",
    "            \n",
    "        if i == 1:\n",
    "            ids_path =  path_ids + \"\\Category_1.txt\" ## Getting path for first category\n",
    "            re2 = open(ids_path,\"a\") ## Creating Category_1.txt to stote category_1 ids\n",
    "            file_open = open(filename,\"r\")\n",
    "            for line in file_open:\n",
    "                line = line.split()\n",
    "                x = int(line[8])\n",
    "                y = int(line[9])\n",
    "                if (x>=25) and (y<=100): ## Category_name: UPSTREAM_3'UTR\n",
    "                    re2.write((\">\"+line[0])) ## Writing >header to Category_1.txt file\n",
    "                    re2.write(\"\\n\")\n",
    "                    index_1 = index_1 + 1 ## Keep traking the count number of hits in Category_1.txt file\n",
    "            print(\"Two\",index_1) ## Printing total hits in Category_1.txt\n",
    "\n",
    "            re2.close()\n",
    "\n",
    "\n",
    "#         elif i == 3:\n",
    "#             ids_path =  path_ids + \"\\Category_3.txt\"\n",
    "#             re3 = open(ids_path,\"a\") \n",
    "#             file_open = open(filename,\"r\")\n",
    "#             for line in file_open:\n",
    "#                 line = line.split()\n",
    "#                 x = int(line[8])\n",
    "#                 y = int(line[9])\n",
    "#                 if (x>=101) and (y<=107):\n",
    "#                     re3.write((\">\"+line[0]))\n",
    "#                     re3.write(\"\\n\")\n",
    "#                     index_3 = index_3 + 1\n",
    "#             print(\"Three\",index_3)\n",
    "#             re3.close()\n",
    "\n",
    "        elif i == 2:\n",
    "            ids_path =  path_ids + \"\\Category_2.txt\"  ## Similar to Category_1.txt\n",
    "            re4 = open(ids_path,\"a\") \n",
    "            file_open = open(filename,\"r\")\n",
    "            for line in file_open:\n",
    "                line = line.split()\n",
    "                x = int(line[8])\n",
    "                y = int(line[9])\n",
    "                if (x>=108) and (y<=329): ## Category_name: DOWNSTREAM_3'UTR\n",
    "                    re4.write((\">\"+line[0]))\n",
    "                    re4.write(\"\\n\")\n",
    "                    index_2 = index_2 + 1\n",
    "            print(\"Four\",index_2)\n",
    "            re4.close()\n",
    "\n",
    "\n",
    "        elif i == 3:\n",
    "            ids_path =  path_ids + \"\\Category_3.txt\"\n",
    "            re5 = open(ids_path,\"a\") \n",
    "            file_open = open(filename,\"r\")\n",
    "            for line in file_open:\n",
    "                line = line.split()\n",
    "                x = int(line[8])\n",
    "                y = int(line[9])\n",
    "                if (x<=24) and (108<=y<=329): ## Category_name: FP+UPSTREAM_3'UTR+BD+ DOWNSTREAM_3'UTR\n",
    "                    re5.write((\">\"+line[0]))\n",
    "                    re5.write(\"\\n\")\n",
    "                    index_3 = index_3 + 1\n",
    "            print(\"Five\",index_3)\n",
    "            re5.close()\n",
    "\n",
    "\n",
    "        elif i == 4:\n",
    "            ids_path =  path_ids + \"\\Category_4.txt\"\n",
    "            re6 = open(ids_path,\"a\") \n",
    "            file_open = open(filename,\"r\")\n",
    "            for line in file_open:\n",
    "                line = line.split()\n",
    "                x = int(line[8])\n",
    "                y = int(line[9])\n",
    "                if (x<=24) and (101<=y<=107): ## Category_name: FP+UPSTREAM_3'UTR+BD\n",
    "                    re6.write((\">\"+line[0]))\n",
    "                    re6.write(\"\\n\")\n",
    "                    index_4 = index_4 + 1\n",
    "            print(\"Six\",index_4)\n",
    "            re4.close()\n",
    "\n",
    "\n",
    "        elif i == 5:\n",
    "            ids_path =  path_ids + \"\\Category_5.txt\"\n",
    "            re7 = open(ids_path,\"a\")\n",
    "            file_open = open(filename,\"r\")\n",
    "            for line in file_open:\n",
    "                line = line.split()\n",
    "                x = int(line[8])\n",
    "                y = int(line[9])\n",
    "                if (x<=24) and (25<=y<=100): ## Category_name: FP+UPSTREAM_3'UTR\n",
    "                    re7.write((\">\"+line[0]))\n",
    "                    re7.write(\"\\n\")\n",
    "                    index_5 = index_5 + 1\n",
    "            print(\"Seven\",index_5)\n",
    "            re7.close()\n",
    "\n",
    "\n",
    "        elif i == 6:\n",
    "            ids_path =  path_ids + \"\\Category_6.txt\"\n",
    "            re8 = open(ids_path,\"a\")\n",
    "            file_open = open(filename,\"r\")\n",
    "            for line in file_open:\n",
    "                line = line.split()\n",
    "                x = int(line[8])\n",
    "                y = int(line[9])\n",
    "                if (25<=x<=100) and (108<=y<=329): ## Category_name: UPSTREAM_3'UTR+BD+ DOWNSTREAM_3'UTR\n",
    "                    re8.write((\">\"+line[0]))\n",
    "                    re8.write(\"\\n\")\n",
    "                    index_6 = index_6 + 1\n",
    "            print(\"Eight\",index_6)\n",
    "            re8.close()\n",
    "\n",
    "\n",
    "        elif i == 7:\n",
    "            ids_path =  path_ids + \"\\Category_7.txt\"\n",
    "            re9 = open(ids_path,\"a\")\n",
    "            file_open = open(filename,\"r\")\n",
    "            for line in file_open:\n",
    "                line = line.split()\n",
    "                x = int(line[8])\n",
    "                y = int(line[9])\n",
    "                if (25<=x<=100) and (101<=y<=107): ## Category_name:UPSTREAM_3'UTR+BD\n",
    "                    re9.write((\">\"+line[0]))\n",
    "                    re9.write(\"\\n\")\n",
    "                    index_7 = index_7 + 1\n",
    "            print(\"Nine\",index_7)\n",
    "            re9.close()\n",
    "\n",
    "\n",
    "        elif i == 8:\n",
    "            ids_path =  path_ids + \"\\Category_8.txt\"\n",
    "            re10 = open(ids_path,\"a\")\n",
    "            file_open = open(filename,\"r\")\n",
    "            for line in file_open:\n",
    "                line = line.split()\n",
    "                x = int(line[8])\n",
    "                y = int(line[9])\n",
    "                if (101<=x<=107) and (108<=y<=329): ## Category_name:BD+ DOWNSTREAM_3'UTR\n",
    "                    re10.write((\">\"+line[0]))\n",
    "                    re10.write(\"\\n\")\n",
    "                    index_8 = index_8 + 1\n",
    "            print(\"Ten\",index_8)\n",
    "            re10.close()\n",
    "            print(\"*********End of Genotype******\")\n",
    "            \n",
    "            \n",
    "## At the end you will get 8 different Categories named Category_1.txt, Category_2,.....So on.... Category_8.txt\n",
    "\n",
    "## Category_1.txt - UPSTREAM_3'UTR\n",
    "## Category_2.txt - DOWNSTREAM_3'UTR\n",
    "## Category_3.txt - FP+UPSTREAM_3'UTR+BD+ DOWNSTREAM_3'UTR\n",
    "## Category_4.txt - FP+UPSTREAM_3'UTR+BD\n",
    "## Category_5.txt - FP+UPSTREAM_3'UTR\n",
    "## Category_6.txt - UPSTREAM_3'UTR+BD+ DOWNSTREAM_3'UTR\n",
    "## Category_7.txt - UPSTREAM_3'UTR+BD\n",
    "## Category_8.txt - BD+ DOWNSTREAM_3'UTR\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Getting fasta sequences in a loop\n",
    "### Retriving fasta sequences for the ids generated for each category for all 20 genotypes\n",
    "###\n",
    "def getting_fasta(genotype,sequence_file):\n",
    "    main_path = 'C:\\\\Users\\\\johnp\\\\Downloads\\\\poly-a_loop\\\\forward_strand\\\\Genotypes\\\\'\n",
    "    for i in range(1,9):\n",
    "\n",
    "        filename  = main_path + genotype +\"\\\\ids\\\\Category_%i.txt\"%i ## Getting path for the each category file\n",
    "        file_ids = open(filename,\"r\") ## Opening the file\n",
    "\n",
    "        filename_fasta = main_path + genotype + \"\\\\fasta\\\\Category_%i.txt\"%i ## Getting path for each category file that contain fasta sequences\n",
    "        re = open(filename_fasta,\"a\")  ## Creating file for category_x.txt to store fasta sequences         \n",
    "        ids = []\n",
    "        fasta = []\n",
    "        for line in file_ids: ### Looping through each header in Category_x.txt file\n",
    "            line = line.strip(\"\\n\") ## Remove \"\\n\" character\n",
    "            ids.append(line) ## Storing header ids in a list\n",
    "        # print(ids)\n",
    "        ## Getting path for initial fasta files for 20 genotypes\n",
    "        main_path_to_sequence_file = \"C:\\\\Users\\\\johnp\\\\Downloads\\\\poly-a_loop\\\\forward_strand\\\\intial_fasta_files\\\\\"\n",
    "        sequence_path =  main_path_to_sequence_file + sequence_file\n",
    "        file_fasta = open(sequence_path,\"r\")\n",
    "        ids_seq_list = []\n",
    "        for line2 in file_fasta: ## Looping through each line in a fasta file\n",
    "            words = line2.split() ## Converting words in each line to a list\n",
    "            if line2[0] == \">\": ## If a line starts with \">\"\n",
    "                    y = line2 ## Store header in y variable\n",
    "            else: ## If line doesn't starts with \">\"\n",
    "                    line2 = line2.strip(\"\\n\") ## There is \"\\n\" character after each header, we are just getting rid of it\n",
    "                    ids_seq_list.append((y.strip(\"\\n\"),line2)) ## List that contains tuple of (>Header and the fasta sequence)\n",
    "        # print(ids_seq_list)\n",
    "\n",
    "        ## Getting fasta sequences after comparing ids\n",
    "        for x in ids: ## Looping through a list that contains ids\n",
    "        #     print(\"Working\")\n",
    "            for y in ids_seq_list: ## Looping through a list that contains a tuple of (ids and sequences)\n",
    "        #         print(\"Working\")\n",
    "                y1 = y[0] ## Storing id value in (ids and sequences) to a variable y1\n",
    "                if (x == y1): ## If an id in ids_list matches ids_in (ids and sequences)\n",
    "    #                 print(x + \"\\n\" + y[1])\n",
    "                    re.write(x) ## Writing header to a file\n",
    "                    re.write(\"\\n\") ## New line\n",
    "                    re.write(y[1]) ## Writing fasta sequence assosiated with that id\n",
    "                    re.write(\"\\n\") ## New line\n",
    "\n",
    "    re.close()\n",
    "    \n",
    "    return 0\n",
    "## At the end you will get 8 different Categories named Category_1.txt, Category_2,.....So on.... Category_8.txt that\n",
    "# will contain fasta sequences\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Calling function named \"getting_fasta\" with the arguments \"Genotype name\" and \"fasta file\" that contain sequences\"\n",
    "getting_fasta(\"singel_hits_plus_strand_e1-05_soc1_blastn_bd_only_1348_50-1_epat\",\"refine_final_1.fa\")\n",
    "getting_fasta(\"singel_hits_plus_strand_e1-05_soc1_blastn_bd_only_1348_50-2_epat\",\"refine_final_2.fa\")\n",
    "getting_fasta(\"singel_hits_plus_strand_e1-05_soc1_blastn_bd_only_1348_67-1_epat\",\"refine_final_3.fa\")\n",
    "getting_fasta(\"singel_hits_plus_strand_e1-05_soc1_blastn_bd_only_1348_67-2_epat\",\"refine_final_4.fa\")\n",
    "getting_fasta(\"singel_hits_plus_strand_e1-05_soc1_blastn_bd_only_1348_69-1_epat\",\"refine_final_5.fa\")\n",
    "getting_fasta(\"singel_hits_plus_strand_e1-05_soc1_blastn_bd_only_1348_69-2_epat\",\"refine_final_6.fa\")\n",
    "getting_fasta(\"singel_hits_plus_strand_e1-05_soc1_blastn_bd_w_deadenylase_1349_46-1_epat\",\"refine_final_7.fa\")\n",
    "getting_fasta(\"singel_hits_plus_strand_e1-05_soc1_blastn_bd_w_deadenylase_1349_46-2_epat\",\"refine_final_8.fa\")\n",
    "getting_fasta(\"singel_hits_plus_strand_e1-05_soc1_blastn_bd_w_deadenylase_1349_58-1_epat\",\"refine_final_9.fa\")\n",
    "getting_fasta(\"singel_hits_plus_strand_e1-05_soc1_blastn_bd_w_deadenylase_1349_58-2_epat\",\"refine_final_10.fa\")\n",
    "getting_fasta(\"singel_hits_plus_strand_e1-05_soc1_blastn_bd_w_deadenylase_1349_68-1_epat\",\"refine_final_11.fa\")\n",
    "getting_fasta(\"singel_hits_plus_strand_e1-05_soc1_blastn_bd_w_deadenylase_1349_68-2_epat\",\"refine_final_12.fa\")\n",
    "getting_fasta(\"singel_hits_plus_strand_e1-05_soc1_blastn_tvn_1348-1_epat\",\"refine_final_13.fa\")\n",
    "getting_fasta(\"singel_hits_plus_strand_e1-05_soc1_blastn_tvn_1348-2_epat\",\"refine_final_14.fa\")\n",
    "getting_fasta(\"singel_hits_plus_strand_e1-05_soc1_blastn_wt_37-1_epat\",\"refine_final_15.fa\")\n",
    "getting_fasta(\"singel_hits_plus_strand_e1-05_soc1_blastn_wt_37-2_epat\",\"refine_final_16.fa\")\n",
    "getting_fasta(\"singel_hits_plus_strand_e1-05_soc1_blastn_wt_50-1_epat\",\"refine_final_17.fa\")\n",
    "getting_fasta(\"singel_hits_plus_strand_e1-05_soc1_blastn_wt_50-2_epat\",\"refine_final_18.fa\")\n",
    "getting_fasta(\"singel_hits_plus_strand_e1-05_soc1_blastn_wt_54-1_epat\",\"refine_final_19.fa\")\n",
    "getting_fasta(\"singel_hits_plus_strand_e1-05_soc1_blastn_wt_54-2_epat\",\"refine_final_20.fa\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
